blockchain/validate_test.go

```diff
@@ -1566,3 +1567,219 @@ func TestCalcTicketReturnAmounts(t *testing.T) {
 		}
 	}
 }
+
+// TestCheckTicketRedeemers ensures that all ticket redeemer consensus rule
+// checks work as expected.
+func TestCheckTicketRedeemers(t *testing.T) {
+	t.Parallel()
+
+	// Generate a slice of ticket hashes to use for tests.
+	numTicketHashes := 9
+	testTicketHashes := make([]chainhash.Hash, numTicketHashes)
+	for i := 0; i < numTicketHashes; i++ {
+		var randHash chainhash.Hash
+		if _, err := rand.Read(randHash[:]); err != nil {
+			t.Fatalf("error reading random hash: %v", err)
+		}
+		testTicketHashes[i] = randHash
+	}
+
+	// Create a default exists missed ticket function.
+	defaultExistsMissedTicket := func(ticket chainhash.Hash) bool {
+		return false
+	}
+
+	tests := []struct {
+		name                     string
+		voteTicketHashes         []chainhash.Hash
+		revocationTicketHashes   []chainhash.Hash
+		winners                  []chainhash.Hash
+		expiringNextBlock        []chainhash.Hash
+		existsMissedTicket       func(ticket chainhash.Hash) bool
+		isTreasuryEnabled        bool
+		isAutoRevocationsEnabled bool
+		wantErr                  error
+	}{
```

Comment:
Updated.  It was formatted this way since there was a comment above blocks of tests indicating if they were with auto revocations disabled or enabled.  I just made that part of the test name instead and updated the formatting.